clients/segments/segment_test.go

```diff
@@ -456,24 +458,24 @@ func TestUpsert(t *testing.T) {
 		ctx := testutils.ContextWithLogger(t)
 		mockClient := segments.NewMockclient(gomock.NewController(t))
 		mockClient.EXPECT().
-			Get(ctx, "uid", gomock.Any()).
+			Get(ctx, "D82a1jdA23a", gomock.Any()).
 			Return(&http.Response{
 				StatusCode: http.StatusOK,
 				Body:       io.NopCloser(strings.NewReader(apiExistingResource)),
 			}, nil)
 		mockClient.EXPECT().
-			Update(ctx, "uid", gomock.Any(), gomock.AssignableToTypeOf(rest.RequestOptions{})).
+			Update(ctx, "D82a1jdA23a", gomock.Any(), gomock.AssignableToTypeOf(rest.RequestOptions{})).
 			Return(&http.Response{
 				StatusCode: http.StatusOK,
 				Body:       io.NopCloser(strings.NewReader(apiResponse)),
 			}, nil).
 			Do(func(_, _, payload any, ro any) {
 				assertVersionParam(t, ro, "2")
-				assertRequestPayload(payload, t, "uid", "2f321c04-566e-4779-b576-3c033b8cd9e9")
+				assertRequestPayload(payload, t, "D82a1jdA23a", "2f321c04-566e-4779-b576-3c033b8cd9e9")
```

Comment:
nit: This check is only for `owner` and `UID`. This won't catch the bug where the given payload is overwritten with `Owner` and `UUID`, and the given payload is lost.